bee-network/src/peer/meta.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::swarm::protocols::gossip::GossipSender;
+
+#[derive(Clone, Debug, Default)]
+pub struct PeerState(Option<GossipSender>);
+
+impl PeerState {
+    pub fn new_connected(gossip_sender: GossipSender) -> Self {
+        Self(Some(gossip_sender))
+    }
+
+    pub fn new_disconnected() -> Self {
+        Self(None)
+    }
+
+    pub fn is_disconnected(&self) -> bool {
+        self.0.is_none()
+    }
+
+    pub fn is_connected(&self) -> bool {
+        self.0.is_some()
+    }
+
+    pub fn set_connected(&mut self, gossip_sender: GossipSender) -> Option<GossipSender> {
```

Comment:
I'm not sure I'd use `set_` here. From the outside, it doesn't look like anything is being 'set' here. Perhaps just `connect`/`disconnect`?

Reply:
`to_connected`/`to_disconnected`? This is an API on a type called `PeerState`, so I would find it weird to say `state.connect`/`state.disconnect` when I really want to describe a state transition.

Reply:
I changed the implementation to an enum, which is the canonical way anyway to represent a set of states. The `set_` prefix should be more logical now. However I want to emphasize on state transition which is why I now use a `to_` prefix instead.